admin/metabox/class-metabox-section-react.php

```diff
@@ -0,0 +1,109 @@
+<?php
+/**
+ * WPSEO plugin file.
+ *
+ * @package WPSEO\Admin
+ */
+
+/**
+ * Generates and displays the React root element for a metabox section.
+ */
+class WPSEO_Metabox_Section_React implements WPSEO_Metabox_Section {
+
+	/**
+	 * Name of the section.
+	 *
+	 * @var string
+	 */
+	public $name;
+
+	/**
+	 * Content to use above the React root node.
+	 *
+	 * @var string
+	 */
+	public $content;
+
+	/**
+	 * Content to use for the link.
+	 *
+	 * @var string
+	 */
+	private $link_content;
+
+	/**
+	 * Link title to use.
+	 *
+	 * @var string
+	 */
+	private $link_title;
+
+	/**
+	 * Class to add to the link.
+	 *
+	 * @var string
+	 */
+	private $link_class;
+
+	/**
+	 * Aria label to use for the link.
+	 *
+	 * @var string
+	 */
+	private $link_aria_label;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param string $name         The name of the section, used as an identifier in the html. Can only contain URL safe characters.
+	 * @param string $link_content The text content of the section link.
+	 * @param string $content      Optional. Content to use above the React root element.
+	 * @param array  $options      Optional link attributes.
+	 */
+	public function __construct( $name, $link_content, $content = '', array $options = array() ) {
+		$this->name = $name;
+		$this->content = $content;
+
+		$default_options = array(
+			'link_title'      => '',
+			'link_class'      => '',
+			'link_aria_label' => '',
+		);
+
+		$options = array_merge( $default_options, $options );
+
+		$this->link_content    = $link_content;
+		$this->link_title      = $options['link_title'];
+		$this->link_class      = $options['link_class'];
+		$this->link_aria_label = $options['link_aria_label'];
+	}
+
+	/**
+	 * Outputs the section link.
+	 */
+	public function display_link() {
+		printf(
+			'<li><a href="#wpseo-meta-section-%1$s" class="wpseo-meta-section-link %2$s"%3$s%4$s>%5$s</a></li>',
+			esc_attr( $this->name ),
+			esc_attr( $this->link_class ),
+			( '' !== $this->link_title ) ? ' title="' . esc_attr( $this->link_title ) . '"' : '',
+			( '' !== $this->link_aria_label ) ? ' aria-label="' . esc_attr( $this->link_aria_label ) . '"' : '',
+			$this->link_content
+		);
+	}
+
+	/**
+	 * Outputs the section content.
+	 */
+	public function display_content() {
+		$html  = '<div id="%1$s" class="wpseo-meta-section">';
+		$html .= $this->content;
+		$html .= '<div id="wpseosnippet" class="wpseosnippet"></div>';
```

Comment:
if this class is going to be reused for all the other metabox sections, I guess it shouldn't use these hardcoded ID and class

Reply:
This is the root element that React will hook into, all the sections of the metabox will be combined via React. There will only be one of these on a page.

Reply:
OK, but these ID and class are also used to scope some CSS (and for some JS). That means that when rendering other sections, there will probably be the need to refactor some CSS and double check or remove some JS.